v2/user_claims_test.go

```diff
@@ -350,3 +350,19 @@ func TestSourceNetworkValidation(t *testing.T) {
 		t.Error("limits should be invalid")
 	}
 }
+
+func TestUserAllowedConnectionTypes(t *testing.T) {
+	ukp := createUserNKey(t)
+	uc := NewUserClaims(publicKey(ukp, t))
+
+	vr := CreateValidationResults()
+	uc.Validate(vr)
+
+	uc.AllowedConnectionTypes = map[string]struct{}{"WEBSOCKET": struct{}{}}
+	vr = CreateValidationResults()
+	uc.Validate(vr)
+
+	uc.AllowedConnectionTypes = map[string]struct{}{"anything": struct{}{}, "WEBSOCKET": struct{}{}}
```

Comment:
unsure what the meaning of an untyped string is - also if we use the constants we don't have to worry about case etc.

Reply:
Will replace with StringList anyway, but the idea was to make sure that we don't do validation. But again, maybe we should, it's just that we would have to do the "update JWT + release + update server JWT dep + release server" when we add a new type. That's maybe ok after all, we do it for other things.

Reply:
Changed this test to encode, then decode and ensure the values that I added to the list are present.